index.ts

```diff
@@ -166,7 +166,20 @@ interface Card {
 }
 
 let rooms: Room[] = [];
-wss.on("request", function (request) {
+wss.on("request", async function (request) {
+  let name: string, id: string;
+  try {
+    let expressReq = request.httpRequest as express.Request;
+    ({name, id} = await new Promise<{name: string, id: string}>((resolve, reject) => {
+      sessionParser(expressReq, {} as express.Response, () => {
+        resolve({name: expressReq.session.name as string, id: expressReq.session.uid as string});
+      })
+    }));
+  } catch (e) {
```

Comment:
Since `reject()` is not called in `new Promise()` there is actually no check whether the session could be retrieved. If it is guaranteed to succeed, no `try`/`catch` needed here, if `expressReq.session` may be undefined it should be checked for explicitly and call `reject` then.

Reply:
Generally speaking it _should_ not be `undefined` but later code may lead to this case. I don't know why I didn't see the opportunity to reject `!expressReq.session || !expressReq.session.name || !expressReq.session.uid` at this point.